### DISCUSSION TITLE
Defining different environments for env variables when using vercel cli to deploy

### DISCUSSION THREAD

**Discussion by HassenHichri** [vercel/vercel#7547] [Help]
Hi, I'm using Vercel cli to deploy my NextJS app as my code is hosted at a self-hosted Gitlab repo. I want to deploy preview deployments with PROD env variables for hotfix branches for example or deploy a branch with different env variables values. Right now, it seems there is no way to have branch specific env variables values using the cli, also there is no command to modify the value of an env variable. What would be a nice workaround for this?

**Comment by leerob** [comment]
**Update:** It's now much easier to deploy to Vercel with any custom CI/CD provider like self-managed GitLab with GitLab Pipelines!

[We've created a guide](https://vercel.com/support/articles/how-can-i-use-gitlab-pipelines-with-vercel) and [an example](https://github.com/vercel/examples/tree/main/ci-cd/gitlab-cicd) using GitLab Pipelines. Hope this helps! 🙏
